tests/functional/test_contract_event.py

```diff
@@ -56,6 +56,16 @@ def assert_receipt_logs(receipt: "ReceiptAPI", num: int):
     assert_receipt_logs(receipt_2, 3)
 
 
+def test_from_receipt_different_contract(
```

Comment:
@fubuloubu I added this test here. It fails on `main` branch and passes with your fix.